alacritty_terminal/src/term/mod.rs

```diff
@@ -281,6 +281,52 @@ impl<'a, C> RenderableCellsIter<'a, C> {
             cursor_style,
         }
     }
+
+    /// Check selection state of a cell.
+    fn is_selected(&self, point: Point) -> bool {
+        let selection = match self.selection {
+            Some(selection) => selection,
+            None => return false,
+        };
+
+        // Point itself is selected
+        if selection.contains(point.col, point.line) {
+            return true;
+        }
+
+        let num_cols = self.grid.num_cols().0;
+        let cell = self.grid[&point];
+
+        // Check if wide char's spacers are selected
+        if cell.flags.contains(Flags::WIDE_CHAR) {
+            let prevprev = point.sub(num_cols, 2, false);
+            let prev = point.sub(num_cols, 1, false);
+            let next = point.add(num_cols, 1, false);
+
+            // Check trailing spacer
+            return selection.contains(next.col, next.line)
+                // Check line-wrapping, leading spacer
+                || (self.grid[&prev].flags.contains(Flags::WIDE_CHAR_SPACER)
+                    && !self.grid[&prevprev].flags.contains(Flags::WIDE_CHAR)
+                    && selection.contains(prev.col, prev.line));
+        }
+
+        // Check if spacer's wide char is selected
+        if cell.flags.contains(Flags::WIDE_CHAR_SPACER) {
+            let prev = point.sub(num_cols, 1, false);
+
+            if self.grid[&prev].flags.contains(Flags::WIDE_CHAR) {
+                // Check previous cell for trailing spacer
+                return self.is_selected(prev);
+            } else {
+                // Check next cell for line-wrapping, leading spacer
+                let next = point.add(num_cols, 1, false);
+                return self.is_selected(next);
```

Comment:
Could you explain to me why do we need recursion?

Reply:
Because I did not want any duplicate code in this already way too complex logic. These codepaths should rarely be hit, so performance isn't really much of an issue.

But basically what we're doing here is that the `WIDE_CHAR` cell itself has full authority on checking if it is selected or not. So the spacers around it just query that cell for the selection information.

Otherwise we'd have to check in the left spacer if either WIDE_CHAR or right spacer is selected, check in the right spacer if WIDE_CHAR or left spacer is selected and then check in the WIDE_CHAR if either left spacer or right spacer are selected.